### PR TITLE
build: drop support for Python 3.7

### PR DESCRIPTION
**Pull request by sisp**
I've dropped support for Python 3.7. Resolves #1251.

### CODE REVIEW

File: copier/tools.py [hunk 1]
```diff
@@ -147,7 +140,8 @@ def handle_remove_readonly(
 ) -> None:
     """Handle errors when trying to remove read-only files through `shutil.rmtree`.
 
-    This handler makes sure the given file is writable, then re-execute the given removal function.
+    On Windows, `shutil.rmtree` does not handle read-only files very well. This handler
+    makes sure the given file is writable, then re-execute the given removal function.
```
Comment (sisp) [inline copier/tools.py:144]: It seems that`shutil.rmtree` doesn't only have this issue under Python <3.8 on Windows. See the Windows CI checks when I had this handler removed: https://github.com/copier-org/copier/actions/runs/5573531416

It seems okay to remove the custom `TemporaryDirectory` class though. I hope that's actually correct and not just uncovered by tests.

**Comment by sisp** [comment]
I think this PR is ready for a review now.

I'm not sure why there are 3 `build (<OS>-latest, 3.7)` jobs, they shouldn't exist because I removed `"3.7"` from the CI matrix. :thinking:

File: copier/tools.py (hunk 1 above)
Comment (pawamoy) [reply copier/tools.py:139]: Is the `handle_remove_readonly` utility still used somewhere else in the code?

**Comment by yajo** [comment]
That was due to repo settings. Fixed now.

File: copier/tools.py (hunk 1 above)
Comment (sisp) [reply copier/tools.py:139]: No, only in [`template.Template._cleanup`](https://github.com/copier-org/copier/blob/a27366bc72927919914fbe09987ffd16518f0ad8/copier/template.py#L213).

**Review by yajo — APPROVED** [review]

**Comment by sisp** [comment]
@yajo Haha, looks like I'm still struggling with the commit change types. :laughing: I was actually wondering whether `build:` was a bad choice because it wouldn't appear in the changelog.

**Pull request merged**